src/server/scripts/EasternKingdoms/zone_westfall.cpp

```diff
@@ -169,11 +170,361 @@ class spell_westfall_quest_credit_jangolode_event : public SpellScript
+                {
+                    if (!creature->IsAlive() || creature->IsInCombat())
+                    {
+                        _events.ScheduleEvent(EVENT_RESUME_MOVE, 0, EVENT_GROUP_OOC);
+                        return;
+                    }
+                    hoboRage = true;
+                    Talk(SAY_PROPERTY_RAGE);
+                    me->RemoveFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_GOSSIP);
+                    me->PauseMovement();
+                    me->CastSpell(creature, SPELL_AGGRO_HOBO, true);
+                    me->SetFacingToObject(creature);
+                }
+                else
+                    _events.ScheduleEvent(EVENT_RESUME_MOVE, 0, EVENT_GROUP_OOC);
+                break;
+            }
+            case EVENT_RESUME_MOVE:
+            {
+                me->ResumeMovement();
+                if (roll_chance_i(50)) // prevent spam
+                {
+                    if (roll_chance_i(70)) // chance to trigger simple talk event
+                        _events.ScheduleEvent(EVENT_TALK, 30s, 80s, EVENT_GROUP_OOC);
+                    else
+                        _events.ScheduleEvent(urand(EVENT_JACKPOT_INTRO, EVENT_PROPERTY_RAGE), 30s, 80s, EVENT_GROUP_OOC);
+                }
+                else
+                    _events.Repeat(30s, 80s);
+                break;
+            }
+            default:
+                break;
+            }
+        }
+
+        if (!hoboRage)
+            if(!UpdateVictim())
+                return;
+
+        DoMeleeAttackIfReady();
+    }
+
+private:
+    EventMap _events;
+    bool bribeFailed;
+    bool hoboRage;
+    bool flee;
+};
+
+class spell_westfall_summon_ragamuffin_looter : public SpellScript
+{
+    PrepareSpellScript(spell_westfall_summon_ragamuffin_looter);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({
+            SPELL_SUMMON_RAGAMUFFIN_LOOTER_1,
+            SPELL_SUMMON_RAGAMUFFIN_LOOTER_2,
+            SPELL_SUMMON_RAGAMUFFIN_LOOTER_3,
+            SPELL_SUMMON_RAGAMUFFIN_LOOTER_4
+            });
+    }
+
+    void HandleScript(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetCaster(), SPELL_SUMMON_RAGAMUFFIN_LOOTER_1, true);
+        GetCaster()->CastSpell(GetCaster(), SPELL_SUMMON_RAGAMUFFIN_LOOTER_2, true);
+        GetCaster()->CastSpell(GetCaster(), SPELL_SUMMON_RAGAMUFFIN_LOOTER_3, true);
+        GetCaster()->CastSpell(GetCaster(), SPELL_SUMMON_RAGAMUFFIN_LOOTER_4, true);
+    }
+
+    void Register() override
+    {
+        OnEffectLaunch += SpellEffectFn(spell_westfall_summon_ragamuffin_looter::HandleScript, EFFECT_0, SPELL_EFFECT_SCRIPT_EFFECT);
+    }
+};
+
+class spell_westfall_aggro_hobo : public SpellScript
+{
+    PrepareSpellScript(spell_westfall_aggro_hobo);
+
+    void HandleDummy(SpellEffIndex /*effIndex*/)
+    {
+        ENSURE_AI(npc_westfall_hobo_witness, GetHitUnit()->ToCreature()->AI())->AggroByHobo(GetCaster());
```

Comment:
Same here. Don't use that. Use SetGUID(uint32 type) to tell the unit what unit it shall attack